### PR TITLE
#881 character check consent form submit template

### PR DESCRIPTION
**Pull request by HalcyonJAC**
**Author checklist**

- [x] Include primary ticket number in title - e.g. "#123 New styling for widget" - and any additional tickets in description
- [x] Fill in the details below and delete as appropriate
- [x] Be proactive in getting your work approved 💪

---
## What's included?
1. Add  email template of character check consent form submit
2. Use a database trigger on `applications`. When application is updated, if the character checks consent form has just been completed, send the confirmation email if it hasn't been sent before

Note: This issue is related to [#881 Send email to candidate when Character Agency report submitted](https://github.com/jac-uk/apply/pull/887).

## Who should test?
✅ Developers

## How to test?

1. Run node script `npm run nodeScript sendCharacterCheckRequests.js` to generate a notification.

    ```
    {
      items: ['xmnLGyEmaQd2w18BnCJP'],
      type: 'submit',
      exerciseMailbox: 'email',
      exerciseManagerName: 'name',
      dueDate: '13/06/2022',
    }
    ```

2. Run `npm run nodeScript testNotification.js` to send email manually.

    ```
      const applicationId = 'xmnLGyEmaQd2w18BnCJP';
      const type = 'submit';
      const exerciseMailbox = 'email';
      const exerciseManagerName = 'email';
      const dueDate = '13/06/2022';
      const email = 'email';
    ```

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
Email format:

```
If you need to respond to this email, please contact us at ((exerciseMailbox))
 
Dear ((applicantName))
 
Thank you for submitting your character check consent form. We will be in touch again in due course.
 
Regards,
((selectionExerciseManager))
```

Demo:

https://user-images.githubusercontent.com/79906532/174025159-bfaac313-98b6-4ecd-8684-71705dd0c69b.mov

## Related permissions
Have permissions been considered for this functionality?
- No permission changes required

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

**Review by tomlovesgithub — APPROVED** [review]
👍

**Review by warrensearle — CHANGES_REQUESTED** [review]
We need to keep `exercises` collection for JAC staff access only.

The `vacancies` collection exists to share data with candidates and the public so specific fields from `exercises` can be copied to `vacancies` if you need, providing they are public-facing.

**Review by warrensearle — CHANGES_REQUESTED** [review]
Almost there, just lets be super cautious about when emails are sent...

Right now email will only be sent when `application.characterChecks.status` changes to `'completed'`.

Lets introduce another field to ensure this only happens the first time the status changes to 'completed'.

File: functions/actions/applications/onUpdate.js [hunk 1]
```diff
@@ -55,6 +56,18 @@ module.exports = (config, firebase, db) => {
 
     if (characterChecksBefore && characterChecksAfter && characterChecksBefore.status && characterChecksAfter.status) {
       if ((characterChecksBefore.status !== characterChecksAfter.status) && characterChecksAfter.status === 'completed') {
+        // send confirmation email if it hasn't been sent before
```
[the remark this answers is not in the document]
Comment (warrensearle) [reply functions/actions/applications/onUpdate.js:59]: Just to be clear, with an object/map like the above we can track other emails too, e.g.:
```
emailLog: {
  applicationSubmitted: timestamp,
  characterCheckSubmitted: timestamp,
  ...
}
```

[the remark this answers is not in the document]
Comment (HalcyonJAC) [reply functions/actions/applications/onUpdate.js:60]: Thank you for your feedback. I've updated `emailLog.characterCheckSubmitted` once the character check consent form is submitted.

**Review by warrensearle — APPROVED** [review]
👍

**Pull request merged**